### PR TITLE
Feat: Add Lark support for notification (#3053)

### PR DESCRIPTION
**Pull request by basuotian**
Signed-off-by: Shuai Tian <tians1@xiaopeng.com>

### Description of your changes

Feat: Add Lark support for notification (#3053)

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer
@FogDong

### CODE REVIEW

**Comment by msftgits** [comment]
[CLA assistant check](https://cla.opensource.microsoft.com/oam-dev/kubevela?pullRequest=3059) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/oam-dev/kubevela?pullRequest=3059) before we can accept your contribution.<br/><br/>:x: basuotian [sign now](https://cla.opensource.microsoft.com/oam-dev/kubevela?pullRequest=3059)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/oam-dev/kubevela?pullRequest=3059) it.</sub>

**Review by FogDong — COMMENTED** [review]
Cool! Thanks for your contribution!

Can you add some examples in `/docs/examples/workflow/notification/app.yaml`?
Also, please sign the cla :)

**Review by FogDong — APPROVED** [review]

File: charts/vela-core/templates/defwithtemplate/notification.yaml [hunk 1]
```diff
@@ -17,6 +17,38 @@ spec:
...
+        				// +usage=name is the name of the secret
+        				name: string
+        				// +usage=key is the key in the secret
+        				key: string
+        			}
+        		}
+        		// +useage=Specify the message that you want to sent
+        		message: {
+        			// +usage=msgType can be text, post, image, interactive, share_chat, share_user, audio, media, file, sticker
+        			msg_type: string
```
Comment (wonderflow) [inline charts/vela-core/templates/defwithtemplate/notification.yaml:35]: msgType vs msg_type

File: charts/vela-core/templates/defwithtemplate/notification.yaml [hunk 2]
```diff
@@ -17,6 +17,38 @@ spec:
...
+        			secretRef: {
+        				// +usage=name is the name of the secret
+        				name: string
+        				// +usage=key is the key in the secret
+        				key: string
+        			}
+        		}
+        		// +useage=Specify the message that you want to sent
+        		message: {
+        			// +usage=msgType can be text, post, image, interactive, share_chat, share_user, audio, media, file, sticker
```
Comment (FogDong) [inline charts/vela-core/templates/defwithtemplate/notification.yaml:34]:
```suggestion
        			// +usage=msg_type can be text, post, image, interactive, share_chat, share_user, audio, media, file, sticker
```

**Review by wonderflow — APPROVED** [review]
generally LGTM, thanks!

File: charts/vela-core/templates/defwithtemplate/notification.yaml (hunk 1 above)
Comment (basuotian) [reply charts/vela-core/templates/defwithtemplate/notification.yaml:35]: Thx for review, according to lark api, the key name should be msg_type, i will fix the annotation from msgType to msg_type

**Pull request merged**